### PR TITLE
[Bug](CTAS): Ctas rollback ignore some case

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Currently, some error are catched due to table can not drop when execute ctas, I add a sessionVarible to control drop or not table.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.17 seconds
 load time: 496 seconds
 storage size: 17122635726 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230130131304_clickbench_pr_87143.html

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 1]
```diff
@@ -1954,18 +1955,27 @@ private void handleCtasStmt() {
             try {
                 parsedStmt = ctasStmt.getInsertStmt();
                 execute();
+                if (MysqlStateType.ERR.equals(context.getState().getStateType())) {
+                    LOG.warn("CTAS insert data error, stmt={}", ctasStmt.toSql());
+                    handleCtasRollback(ctasStmt.getCreateTableStmt().getDbTbl());
+                }
             } catch (Exception e) {
                 e.printStackTrace();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:1963]: remove this e.printStackTrace()

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -691,6 +692,10 @@ public class SessionVariable implements Serializable, Writable {
     @VariableMgr.VarAttr(name = ENABLE_FILE_CACHE, needForward = true)
     public boolean enableFileCache = true;
 
+    // Whether drop table when create table as select insert data appear error.
+    @VariableMgr.VarAttr(name = ENABLE_CTAS_ERROR_DROP_TABLE)
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:696]:
```suggestion
    @VariableMgr.VarAttr(name = ENABLE_CTAS_ERROR_DROP_TABLE needForward=true)
```

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 3]
```diff
@@ -262,6 +262,7 @@ public class SessionVariable implements Serializable, Writable {
     public static final String ENABLE_FILE_CACHE = "enable_file_cache";
 
     public static final String GROUP_BY_AND_HAVING_USE_ALIAS_FIRST = "group_by_and_having_use_alias_first";
+    public static final String ENABLE_CTAS_ERROR_DROP_TABLE = "enable_ctas_error_drop_table";
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:265]: How about:
drop_table_if_ctas_failed

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**